lib/providers/controllerAutoCompleteProvider.js

```diff
@@ -283,6 +282,65 @@ const customAlloyCompletionRules = [
 		}
 	},
 
+	{
+		regExp: /(?:Alloy)?\S+/i,
+		getCompletions(request) {
+			let completions;
+			let line = Utils.getLine(request);
+
+			const regResult = line.match(/((?:Alloy)\.(?:(?:[A-Z]\w*)\.?)*)([a-z]\w*)*$/);
+			if (regResult) {
+
+				let apiName;
+				let attribute;
+				if (regResult && regResult.length === 3) {
+					apiName = regResult[1];
+					if (apiName.lastIndexOf('.') === apiName.length - 1) {
+						apiName = apiName.substr(0, apiName.length - 1);
+					}
+					attribute = regResult[2];
+				}
+
+				completions = [];
```

Comment:
Do we need to ensure that `completions` is undefined if there's no match? Or could we initialise it as an array on line 288?